### PR TITLE
#159250138 Add fitbit support

### PR DESCRIPTION
**Pull request by mutebironald**
#### What does this PR do?
Adds fitbit support

#### Description of Task to be completed?
Being able to pull in  weight information from fitbit.

#### How should this be manually tested?
After cloning this repository;
-  Run the following commands
    - `python manage.py runserver`
    -   login as an admin
    -   click the Administrator bar and select sync with fitbit
    -   click sync with fitbit on the subsequent page
    -   Select all options on the fitbit page

#### What are the relevant pivotal tracker stories?
[#159250138](https://www.pivotaltracker.com/story/show/159250138)

####Screenshots
screen shot 2018-08-10 at 09 33 51

screen shot 2018-08-09 at 15 55 42
screen shot 2018-08-09 at 15 55 57
screen shot 2018-08-09 at 15 55 26

### CODE REVIEW

File: wger/core/views/user.py [hunk 1]
```diff
@@ -556,3 +563,155 @@ def get_context_data(self, **kwargs):
...
+
+@login_required
+def add_fitbit_support(request, code=None):
+    '''
+    Gets data from fitbit upon the user authorizing Wger to access their data
+    '''
+    template_data = {}
+    client_id = "22CY22"
+    client_secret = "25058b02498c0b78ce12249454570331"
+    redirect_uri = "https://wger-aces-staging-pr-20.herokuapp.com/en/user/add_fitbit"
```
[the remark this answers is not in the document]
Comment (mutebironald) [reply wger/core/views/user.py]: @Jpkat92 , let me fix this

**Comment by mutebironald** [comment]
@Jpkat92 ,I have fixed the changes you requested.

**Comment by mutebironald** [comment]
@archbishop12 , I have fixed the errors and removed the migration files

**Comment by isheebo** [comment]
change the scope of your commit to `fitbitIntegration`

File: wger/core/tests/test_fitbit.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from unittest import TestCase
+from fitbit import Fitbit
+import copy
```
Comment (isheebo) [inline wger/core/tests/test_fitbit.py]: sort your imports in a pythonic way. If you are using vscode, right click in this file and select sort imports

File: wger/core/tests/test_fitbit.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from unittest import TestCase
```
Comment (isheebo) [inline wger/core/tests/test_fitbit.py]: prefer the `import unittest` to this ☝️

File: wger/core/views/user.py [hunk 4]
```diff
@@ -15,13 +15,18 @@
 # You should have received a copy of the GNU Affero General Public License
 
 import logging
+import requests
```
Comment (isheebo) [inline wger/core/views/user.py]: sort imports here

File: wger/core/views/user.py [hunk 5]
```diff
@@ -556,3 +564,155 @@ def get_context_data(self, **kwargs):
...
+
+
+@login_required
+def add_fitbit_support(request, code=None):
+    """
+    Get data from fitbit upon the user authorizing Wger to access their data.
+    """
+    template_data = {}
+    client_id = os.environ.get('CLIENT_ID', "22CY22")
+    client_secret = os.environ.get('CLIENT_SECRET', "25058b02498c0b78ce12249454570331")  # noqa F821
```
Comment (isheebo) [inline wger/core/views/user.py]: you can break this onto another line and please avoid silencing `flake8` @mutebironald

Comment (isheebo) [reply wger/core/views/user.py]: cc @cohort-7-aces

File: wger/core/views/user.py [hunk 6]
```diff
@@ -556,3 +564,155 @@ def get_context_data(self, **kwargs):
                                           _('Gym')],
                                  'users': context['object_list']['members']}
         return context
+
+
+@login_required
+def add_fitbit_support(request, code=None):
```
Comment (isheebo) [inline wger/core/views/user.py]: @mutebironald This method is approximately 200 lines long 😱. Challenge yourself. Break it down into constituent components of about 30-40 lines each or less. 😄

Comment (mutebironald) [reply wger/core/views/user.py]: @isheebo , Its a challenge indeed.

File: wger/core/views/user.py (hunk 4 above)
Comment (mutebironald) [reply wger/core/views/user.py]: sorted

File: wger/core/tests/test_fitbit.py (hunk 2 above)
Comment (mutebironald) [reply wger/core/tests/test_fitbit.py]: sorted

**Comment by magicmarie** [comment]
@mutebironald Travis build is failing. Fix it

**Pull request merged**